### PR TITLE
DOCS-284 - Updated tiles documentation

### PR DESCRIPTION
**Pull request by jimmyjames**
This also addresses tickets DOCS-229, DOCS-232, and DOCS-230.

This is a large rewrite and expansion of the Device Handler tiles docs.

The changes include:
- Reorganization of content
- Images updated to be consistent with the new mobile app UX
- Expanded documentation of the different control tile types (color and slider)
- Documented the color control's return values
- Split out each type of Multi-Attribute tile into its own section
- Added examples for reference and experimentation.

This commit does not include documentation on the Video tile. That work is captured in DOCS-296.

@pennig I'd like you to review this (it's a lot, so thank you in advance!) and/or delegate as needed.

/cc @unixbeast @mrnohr @KyleLeneau

### CODE REVIEW

File: device-type-developers-guide/tiles-metadata.rst [hunk 1]
```diff
@@ -1,251 +1,501 @@
...
+^^^^^^^^^^^^
+
+We can hard-code a label for state values, or use the state name or current value of the attribute.
+The following label values can be used to display real-time information about the device:
+
+============================= ============
+Label                         Description
+============================= ============
+``label: '${currentValue}'``  The current value of this attribute's state. This is used when the attribute doesn't have a discrete value set, like temperature or power.
+``label: '${name}'``          The name of the attribute state. This is useful when the attribute state is a discreet value, like "on" or "off".
```
Comment (pennig) [inline device-type-developers-guide/tiles-metadata.rst]: discrete

File: device-type-developers-guide/tiles-metadata.rst [hunk 2]
```diff
@@ -299,192 +586,445 @@ Many of the camera Device Handlers will make use of the ``carouselTile()``.
...
-COLOR_CONTROL        Color palette button                  .. image:: ../img/device-types/COLOR_CONTROL.png
-VALUE_CONTROL        Up and down buttons (see note below)  .. image:: ../img/device-types/VALUE_CONTROL.png
-===================  ====================================  =====================================================
+You may also see a ``level`` and ``alpha`` attribute returned from the color control.
+These values are not controlled by the color palette, so are not useful.
 
 .. note::
 
-  The color of the multi-attribute tile is controlled by the PRIMARY_CONTROL tile attribute, or in the case of a ``"thermostat"`` type, the OPERATING_STATE attribute will be used (see below). It will default to a light gray color. If the PRIMARY_CONTROL attribute contains states that change the color, the color of the multi attribute tile will also change.
+    You may some code for Color Control bulbs that adjusts the Hue using some magic numbers and fun math.
```
Comment (pennig) [inline device-type-developers-guide/tiles-metadata.rst]: "You may see code…"?

Comment (jimmyjames) [inline device-type-developers-guide/tiles-metadata.rst]: yep that makes more sense :)

File: device-type-developers-guide/tiles-metadata.rst [hunk 3]
```diff
@@ -299,192 +586,445 @@ Many of the camera Device Handlers will make use of the ``carouselTile()``.
...
+===================  ===========
+Key                  Description
+===================  ===========
+``PRIMARY_CONTROL``  Necessary to render the background of the tile.
+``MEDIA_STATUS``     Used to display and control the current play status (playing, paused, stopped)
+``PREVIOUS_TRACK``   Renders a control for going to the previous track
+``NEXT_TRACK``       Renders a control for going to the next track
+``SLIDER_CONTROL``   Renders a control to select a volume level
+``MEDIA_MUTED``      Allows the user to press the volume icon to mute
+``MARQUEE``          Will display the currently playing track description below the ``PRIMARY_CONTROL``
```
Comment (pennig) [inline device-type-developers-guide/tiles-metadata.rst]: Add to marquee explanation: "Use newlines to delimit fields such as title, artist, album, etc."

File: device-type-developers-guide/tiles-metadata.rst [hunk 4]
```diff
@@ -299,192 +586,445 @@ Many of the camera Device Handlers will make use of the ``carouselTile()``.
...
+.. tip::
+
+    Check out the :ref:`tiles_examples` to see it in action!
+
+.. _generic_multi_attr_tile:
+
+Generic Multi-Attribute Tile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If none of the predefined Multi-Attribute Tile types fit our needs, we can use the Generic Multi-Attribute Tile.
```
Comment (pennig) [inline device-type-developers-guide/tiles-metadata.rst]: s/our/your/

File: device-type-developers-guide/tiles-metadata.rst [hunk 5]
```diff
@@ -299,192 +586,445 @@ Many of the camera Device Handlers will make use of the ``carouselTile()``.
...
+The supported tile attribute types for the Generic Multi-Attribute Tile are shown in the following table:
```
Comment (pennig) [inline device-type-developers-guide/tiles-metadata.rst]: The table here doesn't actually render properly (i.e. at all). Not sure what's wrong...

**Comment by jimmyjames** [comment]
New commits made based on review feedback

**Comment by pennig** [comment]
👍

**Pull request merged**